store/tikv/unionstore/memdb.go

```diff
@@ -259,12 +260,12 @@ func (db *MemDB) GetValueByHandle(handle MemKeyHandle) ([]byte, bool) {
 
 // Len returns the number of entries in the DB.
 func (db *MemDB) Len() int {
-	return db.count
+	return int(atomic.LoadInt64(&db.count))
```

Comment:
We could use the memdb benchmark tests to verify the possible regression using atomic.

Reply:
Benchmark result:
| MemDB        | Master (ns/op) | PR            |
| ------------ | -------------- | ------------- |
| LargeIndex   | 3,948,553,195  | 4,004,971,121 |
| Put          | 442.34         | 447.26        |
| PutRandom    | 1058.6         | 1057.6        |
| Get          | 216.0          | 219.8         |
| GetRandom    | 884.68         | 903.24        |
| BufferSeq    | 35,574,137     | 35,617,929    |
| BufferRandom | 71,613,951     | 70,913,784    |
| Iter         | 1,600,097      | 1,590,371     |
| Creation     | 23.35          | 23.63         |

Which IMO is acceptable.